stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/common/SingleLiveEvent.kt

```diff
@@ -0,0 +1,45 @@
+package io.getstream.chat.ui.sample.common
+
+import androidx.annotation.MainThread
+import androidx.annotation.Nullable
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.Observer
+import io.getstream.chat.android.client.logger.ChatLogger
+import java.util.concurrent.atomic.AtomicBoolean
+
+class SingleLiveEvent<T> : MutableLiveData<T>() {
+
+    private val logger = ChatLogger.get("SingleLiveEvent")
+    private val mPending = AtomicBoolean(false)
+
+    override fun observe(owner: LifecycleOwner, observer: Observer<in T>) {
+        if (hasActiveObservers()) {
+            logger.logD("Multiple observers registered but only one will be notified of changes.")
+        }
```

Comment:
We should never have this fail silently like this. Instead, either support multiple observers, or explicitly crash if someone attempts to register a second observer. Debugging a connected observer that doesn't receive events is more than painful.